src/nonempty.rs

```diff
@@ -0,0 +1,113 @@
+use {Rect, Box2D, Box3D, size2, point2, point3};
+use approxord::{min, max};
+use num::Zero;
+use core::ops::Deref;
+use core::ops::{Add, Sub};
+use core::cmp::{PartialEq};
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
could be useful to serialize it transparently, i.e. `#[serde(transparent)]`

Reply:
Good idea, done.